packages/eslint-plugin/tests/rules/consistent-type-definitions.test.ts

```diff
@@ -0,0 +1,155 @@
+import rule from '../../src/rules/consistent-type-definisions';
+import { RuleTester } from '../RuleTester';
+
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+});
+
+ruleTester.run('consistent-type-definisions', rule, {
+  valid: [
+    `var foo = { };`,
+    `interface A {}`,
+    `interface A extends B { x: number; }`,
+    `type U = string;`,
+    `type V = { x: number; } | { y: string; };`,
+    `
+type Record<T, U> = {
+    [K in T]: U;
+}
+`,
+  ],
```

Comment:
you should also add tests which set the option to one or the other, to validate your options work as expected
i.e.
```ts
{
  code: `type A = { x: number }`,
  options: ['type'],
}
```